core/lib/zksync_core/src/consensus/testonly.rs

```diff
@@ -0,0 +1,291 @@
+use crate::genesis::{ensure_genesis_state, GenesisParams};
+use crate::state_keeper::tests::{
+    create_l1_batch_metadata, create_l2_transaction, MockBatchExecutorBuilder,
+};
+use crate::state_keeper::{MiniblockSealer, ZkSyncStateKeeper};
+use crate::sync_layer::{
+    sync_action::{ActionQueue, ActionQueueSender, SyncAction},
+    ExternalIO, MainNodeClient, SyncState,
+};
+use anyhow::Context as _;
+use rand::Rng;
+use zksync_concurrency::{ctx, scope, sync, time};
+use zksync_consensus_roles::validator;
+use zksync_contracts::SystemContractCode;
+use zksync_dal::ConnectionPool;
+use zksync_types::{
+    api, Address, L1BatchNumber, L2ChainId, MiniblockNumber, ProtocolVersionId, H256,
+};
+
+#[derive(Debug)]
+struct MockMainNodeClient;
```

Comment:
Nit: You can probably reuse `MockMainNodeClient` from `sync_layer::tests` here; w/o generating blocks for it, it effectively works in the same way.

Reply:
done